packages/go/analysis/ad/esc10.go

```diff
@@ -0,0 +1,150 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package ad
+
+import (
+	"context"
+	"errors"
+	"github.com/specterops/bloodhound/analysis"
+	"github.com/specterops/bloodhound/analysis/impact"
+	"github.com/specterops/bloodhound/dawgs/cardinality"
+	"github.com/specterops/bloodhound/dawgs/graph"
+	"github.com/specterops/bloodhound/dawgs/ops"
+	"github.com/specterops/bloodhound/dawgs/query"
+	"github.com/specterops/bloodhound/dawgs/util/channels"
+	"github.com/specterops/bloodhound/graphschema/ad"
+	"github.com/specterops/bloodhound/log"
+)
+
+func PostADCSESC10a(ctx context.Context, tx graph.Transaction, outC chan<- analysis.CreatePostRelationshipJob, groupExpansions impact.PathAggregator, eca, domain *graph.Node, cache ADCSCache) error {
+	results := cardinality.NewBitmap32()
+
+	if canAbuseUPNRels, err := FetchCanAbuseUPNCertMappingRels(tx, eca); err != nil {
+		if graph.IsErrNotFound(err) {
+			return nil
+		}
+
+		return err
+	} else if len(canAbuseUPNRels) == 0 {
+		return nil
+	} else if publishedCertTemplates, ok := cache.PublishedTemplateCache[eca.ID]; !ok {
+		return nil
+	} else {
+		for _, template := range publishedCertTemplates {
+			if valid, err := isCertTemplateValidForESC10a(template); err != nil {
+				if !errors.Is(err, graph.ErrPropertyNotFound) {
+					log.Errorf("Error checking cert template validity for template %d: %v", template.ID, err)
+				} else {
+					log.Debugf("Error checking cert template validity for template %d: %v", template.ID, err)
+				}
+			} else if !valid {
+				continue
+			} else if certTemplateControllers, ok := cache.CertTemplateControllers[template.ID]; !ok {
+				log.Debugf("Failed to retrieve controllers for cert template %d from cache", template.ID)
+				continue
+			} else if ecaControllers, ok := cache.EnterpriseCAEnrollers[eca.ID]; !ok {
+				log.Debugf("Failed to retrieve controllers for enterprise ca %d from cache", eca.ID)
+				continue
+			} else {
+				//Expand controllers for the eca + template completely because we don't do group shortcutting here
+				var (
+					victimBitmap = expandNodeSliceToBitmapWithoutGroups(certTemplateControllers, groupExpansions)
+					ecaBitmap    = expandNodeSliceToBitmapWithoutGroups(ecaControllers, groupExpansions)
+				)
+				victimBitmap.And(ecaBitmap)
+				//Use our id list to filter down to users
+				if userNodes, err := ops.FetchNodeSet(tx.Nodes().Filterf(func() graph.Criteria {
+					return query.And(
+						query.KindIn(query.Node(), ad.User),
+						query.InIDs(query.NodeID(), cardinality.DuplexToGraphIDs(victimBitmap)...),
+					)
+				})); err != nil {
+					if !graph.IsErrNotFound(err) {
+						return err
```

Comment:
Do we want to exit out of the 10a function here or just `continue` to the next template in the loop?

Reply:
Probably a good idea, I'll fix it